### PR TITLE
Attendance service delete by id function

### PR DESCRIPTION
**Pull request by eriadhami**

### CODE REVIEW

File: OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validation.Delete.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+﻿// ---------------------------------------------------------------
```
Comment (tobidub) [inline OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validation.Delete.cs:1]: Classname should be "AttendanceServiceTests.Validations.Delete.cs" or?

Comment (eriadhami) [reply OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validation.Delete.cs]: @tobidub, @hassanhabib  thanks. Changed that.

File: OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Logic.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        [Fact]
+        public async Task ShouldDeleteAttendanceAsync()
+        {
+            // given
+            DateTimeOffset dateTime = GetRandomDateTime();
+            Attendance randomAttendance = CreateRandomAttedance(dateTime: dateTime);
+            Guid inputAttendanceId = randomAttendance.Id;
+            Attendance inputAttendance = randomAttendance;
+            Attendance storageAttendance = randomAttendance;
+            Attendance expectedAttendance = randomAttendance;
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Logic.cs:62]: You should pass the name forward as in:
```
randomThing = CreateRandomThing();
inputThing = randomThing;
storageThing = inputThing (unless you are on modify then clone it from input)
expectedThing = storageThing;

```

File: OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Delete.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        {
+            // given
+            Guid randomAttendanceId = default;
+            Guid inputAttendanceId = randomAttendanceId;
+
+            var invalidAttendanceInputException = new InvalidAttendanceInputException(
+                parameterName: nameof(Attendance.Id),
+                parameterValue: inputAttendanceId);
+
+            var expectedAttendanceValidationException = new AttendanceValidationException(invalidAttendanceInputException);
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Delete.cs:28]: is this line under 120 characters?

File: OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Retrieve.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnRetrieveWhenStorageAttendanceIsNullAndLogItAsync()
+        {
+            // given
+            DateTimeOffset dateTimeOffset = GetRandomDateTime();
+            Guid randomAttendanceId = Guid.NewGuid();
+            Guid inputAttendanceId = randomAttendanceId;
+            Attendance nullAttendance = default;
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Retrieve.cs:64]: Make this as follows:
```csharp

Attendance noAttendance = null;
```

**Comment by hassanhabib** [comment]
@eriadhami please make sure you address any conflicts as well

File: OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Logic.cs (hunk 2 above)
Comment (eriadhami) [reply OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Logic.cs]: > You should pass the name forward as in:
> 
> ```
> randomThing = CreateRandomThing();
> inputThing = randomThing;
> storageThing = inputThing (unless you are on modify then clone it from input)
> expectedThing = storageThing;
> ```

@hassanhabib thanks bro

File: OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Delete.cs (hunk 3 above)
Comment (eriadhami) [reply OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Delete.cs]: > is this line under 120 characters?

@hassanhabib thanks bro

File: OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Retrieve.cs (hunk 4 above)
Comment (eriadhami) [reply OtripleS.Web.Api.Tests.Unit/Services/AttendanceServiceTests/AttendanceServiceTests.Validations.Retrieve.cs]: > Make this as follows:
> 
> ```cs
> Attendance noAttendance = null;
> ```

@hassanhabib thanks bro

**Comment by eriadhami** [comment]
> @eriadhami please make sure you address any conflicts as well

@hassanhabib thanks bro. Everything done

**Pull request merged**